engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreHeartbeatWriteActor.scala

```diff
@@ -38,18 +61,59 @@ case class WorkflowStoreHeartbeatWriteActor(workflowStoreAccess: WorkflowStoreAc
   override def commandToData(snd: ActorRef): PartialFunction[Any, (WorkflowId, OffsetDateTime)] = {
     case command: WorkflowStoreWriteHeartbeatCommand => (command.workflowId, command.submissionTime)
   }
+
+  /*
+  WARNING: Even though this is in an actor, the logic deals with instances of Future that could complete in _any_ order,
+  and even call this method at the same time from different threads.
+
+  We are expecting the underlying FSM to ensure that the call to this method does NOT occur in parallel, waiting for
+  the call to `process` to complete.
+   */
+  private def trackRepeatedFailures(heartbeatDateTime: OffsetDateTime, workflowCount: Int)(processTry: Try[Int]): Unit = {
+    processTry match {
+      case Success(_) =>
+        lastSuccessOption = Option(heartbeatDateTime)
+      case Failure(_: Exception) =>
+        val lastSuccess = lastSuccessOption.getOrElse(
+          sys.error("The lastSuccessOption was somehow still None when calling trackRepeatedFailures")
```

Comment:
This doesn't seem too crazy... what if there's something wrong with Cromwell or the DB that causes all heartbeat writes to fail from the very beginning?